sstables/sstables.cc

```diff
@@ -2414,7 +2414,7 @@ static entry_descriptor make_entry_descriptor(sstring sstdir, sstring fname, sst
     } else {
         throw malformed_sstable_exception(seastar::format("invalid version for file {}. Name doesn't match any known version.", fname));
     }
-    return entry_descriptor(sstdir, ks, cf, generation_from_value(boost::lexical_cast<unsigned long>(generation)), version, format_from_string(format), sstable::component_from_sstring(version, component));
+    return entry_descriptor(sstdir, ks, cf, generation_from_value(boost::lexical_cast<sstables::generation_type::int_t>(generation)), version, format_from_string(format), sstable::component_from_sstring(version, component));
 }
```

Comment:
s -> int_t -> value_t ?

Reply:
currently it's still only int.
value_type will be a std::variant of int_t and sstable_id ( a tagged uuid).
let's change it then

Reply:
yeah, we should revisit all the `int_t` then, and 

- either generalize it to `value_t`, or
- keep on using `int_t` if the use case only applies to `int_t`, or add 
- yet another case to handle the uuid case.

Reply:
ahh, quite a few of `int_t` have been replaced with `generation_type`, so 2 out of 3 action items above have been completed!